src/jobs/helm_push.yaml

```diff
@@ -0,0 +1,39 @@
+description: "Set version and publish to chart museum"
+
+executor: aws
+
+parameters:
+  chart_name:
+    type: string
+    description: "name of the helm chart"
+  helm_dir:
+    type: string
+    description: "directory containing the helm chart"
+    default: ".helm"
+  charts_repo_url:
+    type: string
+    description: "charts repo domain ssm parameter"
+    default: "https://charts.codacy.com"
+
+environment:
+  CHART_NAME: << parameters.chart_name >>
+  HELM_DIR: << parameters.helm_dir >>
+
+steps:
+  - attach_workspace:
+      at: ~/
+  - run:
+      name: Set image tag and version using semver
+      command: |
+        ytool -s appVersion "$(cat .version)" -s version "$(cat .version)" -f "${HELM_DIR}/${CHART_NAME}/Chart.yaml" -e
```

Comment:
instead of mangling the yml files can we enforce some good practice on the Helm files themselves to inject these vaules, or alternatively to use/build a simple helm plugin to handle this requirement?

Reply:
What is the problem with this approach?

Reply:
Right now we just want a way to bump versions of an helm chart before release. This approach solves our problem by treating the file for what it is.. a yml file.

In my point of view a simple `sed` in here would do it just fine, but I appreciate the extra-mile of having a tool that does this specific for yaml (without loosing the structure and comments).

Reply:
The problem is that we are mangling source code in CI before pushing a stable artifact, in this way we loose the source of truth.

Can we inject the version by leveraging Jinja templates?
I appreciate the effort of not using a plain `sed` but I'm confident we have other approaches to the problem itself.
cc. @rtfpessoa

Reply:
Helm dosen't use jinja templates. We would have to add a new tool for that, this tool would be more complicated than the one we are using here. Also it wouldn't solve the problem you stated:

>  mangling source code in CI before pushing a stable artifact, in this way we loose the source of truth. 

There is no way of having the source of truth for the version in the repo without commiting and push.

Do you have a proposal to solve this problem?

Reply:
in the other repository we avoid to store the version in the files at all, and we inject in CI, whatever method you find to do it is fine to me.